libraries/Microsoft.Bot.Builder/Inspection/InspectionSession.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Bot.Builder
+{
+    public class InspectionSession
+    {
+        private readonly ConversationReference _conversationReference;
+        private readonly MicrosoftAppCredentials _credentials;
+        private readonly HttpClient _httpClient;
+        private readonly ILogger _logger;
+
+        public InspectionSession(ConversationReference conversationReference, MicrosoftAppCredentials credentials, HttpClient httpClient, ILogger logger)
+        {
+            _conversationReference = conversationReference;
+            _credentials = credentials;
+            _httpClient = httpClient;
+            _logger = logger;
+        }
+
+        public Task<bool> SendAsync(Activity activity, CancellationToken cancellationToken)
+        {
+            activity.ChannelId = _conversationReference.ChannelId;
+            activity.ServiceUrl = _conversationReference.ServiceUrl;
```

Comment:
Do you have to clone the activity before you do this? The mutation here seems odd, as the *caller* will see these.

Reply:
this code now dead